diff --git a/dist/assets/utils/transformToPath.js b/dist/assets/utils/transformToPath.js
index 22057cb412da95f5cb764d5e26e683b685f2b5f8..87adf548dc5714713b70cf7566741510b45b6897 100644
--- a/dist/assets/utils/transformToPath.js
+++ b/dist/assets/utils/transformToPath.js
@@ -7,7 +7,7 @@ const INVALID_CHAR_REGEX = /[\u0000-\u001F"#$%&*+,:;<=>?[\]^`{|}\u007F]/g;
 function propsToFilename(filePath, transform, hash) {
   let filename = decodeURIComponent(removeQueryString(filePath));
   const ext = extname(filename);
-  if (filePath.startsWith("data:")) {
+  if (filePath.startsWith("data:") || filePath.length > 30) {
     filename = shorthash(filePath);
   } else {
     filename = basename(filename, ext).replace(INVALID_CHAR_REGEX, "_");
