diff --git a/dist/assets/utils/hash.js b/dist/assets/utils/hash.js
index 171f39d58241d3dad4f32ab630cd9c6d4855f3fc..59033e2532ea7d700e3d314e8e80b155a89449aa 100644
--- a/dist/assets/utils/hash.js
+++ b/dist/assets/utils/hash.js
@@ -29,7 +29,7 @@ function extname(filePath) {
 function propsToFilename(filePath, transform, hash) {
   let filename = decodeURIComponent(removeQueryString(filePath));
   const ext = extname(filename);
-  if (filePath.startsWith("data:")) {
+  if (filePath.startsWith("data:") || filePath.length > 30) {
     filename = shorthash(filePath);
   } else {
     filename = basename(filename, ext).replace(INVALID_CHAR_REGEX, "_");
